### PR TITLE
Update OTEL Docs

### PR DESCRIPTION
**Pull request by luneo7**
Updates OpenTelemetry docs to reflect latest changes in the extension.

### CODE REVIEW

File: extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/TracerConfig.java [hunk 1]
```diff
@@ -20,16 +20,6 @@
     /** Build / static runtime config for span exporters */
     public SpanExporterConfig exporter;
 
-    /**
```
Comment (luneo7) [inline extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/TracerConfig.java:23]: Somehow this leftover was merged, and now I am removing it =/

File: docs/src/main/asciidoc/opentelemetry.adoc [hunk 2]
```diff
@@ -210,6 +210,103 @@ Then visit the http://localhost:16686[Jaeger UI] to see the tracing information.
...
 
+== Additional configuration
+Some use cases will require custom configuration of OpenTelemetry.
+These sections will outline what is necessary to properly configure it.
+
+=== ID Generator
+The OpenTelemetry extension will use by default a random https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#id-generators[ID Generator]
+when creating the trace and span identifier.
+
+Some vendor-specific protocols needs a custom ID Generator, you can override the default one by creating a producer.
```
Comment (kenfinnigan) [inline docs/src/main/asciidoc/opentelemetry.adoc:221]: Could you change "needs" to "need"?

File: docs/src/main/asciidoc/opentelemetry.adoc [hunk 3]
```diff
@@ -210,6 +210,103 @@ Then visit the http://localhost:16686[Jaeger UI] to see the tracing information.
...
+
+=== Propagators
+OpenTelemetry propagates cross-cutting concerns through https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md[propagators] that will share an underlying `Context` for storing state and accesing
+data across the lifespan of a distributed transaction.
+
+By default, the OpenTelemetry extension enables the https://www.w3.org/TR/trace-context/[W3C Trace Context] and the https://www.w3.org/TR/baggage/[W3C Baggage]
+propagators, you can however choose any of the supported OpenTelemetry propagators by setting the `propagators` config that is described in the <<configuration-reference>>.
+
+[NOTE]
+====
```
Comment (kenfinnigan) [inline docs/src/main/asciidoc/opentelemetry.adoc:246]: Should we explicitly provide examples of those dependencies to make it easier to add to a `pom.xml`?

**Review by kenfinnigan — CHANGES_REQUESTED** [review]
A couple of minor comments otherwise looks great.

Thanks!

**Comment by kenfinnigan** [comment]
@luneo7 FYI, JDK 11 build failed because the imports aren't sorted.

Can you do an `mvn package` on the extension to fix it? Thanks

File: docs/src/main/asciidoc/opentelemetry.adoc (hunk 3 above)
Comment (luneo7) [reply docs/src/main/asciidoc/opentelemetry.adoc:246]: Yup =] added

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**